proteus/LinearAlgebraTools.py

```diff
@@ -242,15 +325,45 @@ def SparseMatFromDict(nr,nc,aDict):
 
 
 def SparseMat(nr,nc,nnz,nzval,colind,rowptr):
-    """
-    Build a nr x nc sparse matrix from the CSR data structures
-    """
-    import superluWrappers
+    """ Build a nr x nc sparse matrix from the CSR data structures
+
+    Parameters
+    ----------
+    nr : int
+        The number of rows.
+    nc : int
+        The number of columns.
+    nnz : int
+        The number of non-zero matrix entries.
+    nzval : numpy array
+        Array with non-zero matrix entries.
+    colind : numpy array of 32bit integers
+        CSR column array.
+    rowptr : numpy array of 32bit integers
+        CSR row pointer.
+
+    Returns
+    -------
+    sparse_matrix : :class:`proteus.superluWrappers.SparseMatrix`
+        superlu sparse matrix in CSR format.
+
+    Note
+    ----
+    For the superluWrapper, both the colind and rowptr should use
+    32-bit integer data types.
+    """
+    if (colind.dtype != 'int32' or rowptr.dtype != 'int32'):
```

Comment:
Should be logEvent instead of print. could raise TypeError or ValueError maybe instead  of sys.exit